homeassistant/components/smappee/__init__.py

```diff
@@ -75,8 +75,22 @@ async def async_setup(hass: HomeAssistant, config: dict):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Smappee from a zeroconf or config entry."""
     if CONF_IP_ADDRESS in entry.data:
-        smappee_api = api.api.SmappeeLocalApi(ip=entry.data[CONF_IP_ADDRESS])
-        smappee = Smappee(api=smappee_api, serialnumber=entry.data[CONF_SERIALNUMBER])
+        if entry.data[CONF_SERIALNUMBER].startswith("50"):
```

Comment:
It would be slightly cleaner to ask the pypi code if the serial number is a new or old one instead of teaching home assistant directly

Reply:
Changed to helper methode usage from dependency library